modules/watson-machine-learning/main.tf

```diff
@@ -10,6 +10,22 @@ module "cos" {
   cos_plan          = "standard"
 }
 
+module "storage_delegation" {
+  providers = {
+    ibm                           = ibm
+    ibm.deployer                  = ibm
+    restapi.restapi_watsonx_admin = restapi.restapi_watsonx_admin
+  }
+  source               = "github.com/terraform-ibm-modules/terraform-ibm-watsonx-saas-da.git//storage_delegation?ref=v1.4.0"
+  depends_on           = [module.cos]
+  count                = var.cos_kms_crn == null || var.cos_kms_crn == "" ? 0 : 1
```

Comment:
This should be based on the passed watsonx_project_delegated parameter for consistency.